tests/v2/actions/rbac/rbac.go

```diff
@@ -0,0 +1,101 @@
+package rbac
+
+import (
+	"fmt"
+	"strings"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	rbacapi "github.com/rancher/shepherd/extensions/kubeapi/rbac"
+	"github.com/rancher/shepherd/extensions/rbac"
+	"github.com/rancher/shepherd/extensions/users"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Role string
+
+const (
+	Admin                     Role = "admin"
+	RestrictedAdmin           Role = "restricted-admin"
+	StandardUser              Role = "user"
+	ClusterOwner              Role = "cluster-owner"
+	ClusterMember             Role = "cluster-member"
+	ProjectOwner              Role = "project-owner"
+	ProjectMember             Role = "project-member"
+	CreateNS                  Role = "create-ns"
+	ReadOnly                  Role = "read-only"
+	CustomManageProjectMember Role = "projectroletemplatebindings-manage"
+	ActiveStatus                   = "active"
+	ForbiddenError                 = "403 Forbidden"
+	DefaultNamespace               = "fleet-default"
+)
+
+var prtb = v3.ProjectRoleTemplateBinding{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "",
+		Namespace: "",
+	},
+	ProjectName:       "",
+	RoleTemplateName:  "",
+	UserPrincipalName: "",
+}
+
+func (r Role) String() string {
+	return string(r)
+}
+
+// CreateUserAndAddToRole creates a user based on the global role and then adds the user to cluster with provided permissions.
```

Comment:
What about naming it CreateUserAndAddRole? I chose not to name it CreateUserAndAddClusterRole since we're using the same function for adding user as project owner/member.

Reply:
Based on our discussion offline, renamed this to AddUserWithRoleToCluster

Reply:
How about AddUserWithRoleToCluster? I ask because CreateUserAndAssignRole is throwing me as it can also mean adding to another global role.